sdk/src/ConditionalVaultClient.ts

```diff
@@ -73,18 +75,20 @@ export class ConditionalVaultClient {
   ) {
     const storedVault = await this.getVault(vault);
 
-    return (
-      this.mintConditionalTokensIx(
+    try {
+      const tokenDecimals = await getTokenDecimals(storedVault.underlyingTokenMint);
```

Comment:
you can just use storedVault.decimals for this and don't need to make the extra call